contracts/Datastore.sol

```diff
@@ -170,13 +171,35 @@ contract Datastore {
     } 
 
     /**
-     * @notice Delete file with Id `_fileId`
+     * @notice Set file `_fileId` as deleted or not.
      * @param _fileId File Id
+     * @param _isDeleted Is file deleted or not
      */
-    function deleteFile(uint _fileId) public onlyFileOwner(_fileId) {
-        fileList.deleteFile(_fileId);
+    function deleteFile(uint _fileId, bool _isDeleted) public onlyFileOwner(_fileId) {
```

Comment:
This function could receive a `deletePermanently` and an array of fileIds `fileIds` parameters instead of having this function + `deleteFilePermanently` and `deleteFilesPermanently`. This would save us some well needed bytecode :)